source/common/unified_memory_allocation/include/uma/memory_pool.h

```diff
@@ -108,6 +108,13 @@ void umaPoolFree(uma_memory_pool_handle_t hPool, void *ptr);
 ///         returned indicates that the adapter specific result is an error.
 enum uma_result_t umaPoolGetLastResult(uma_memory_pool_handle_t hPool, const char **ppMessage);
 
+///
+/// \brief returns memory pool associated with a given ptr. Memory pointed to by ptr
+///        must have been tracked using provider obtained by calling umaTrackingMemoryProviderCreate
+/// \param ptr pointer to memory belonging to memory pool
+/// \return handle to memory pool that contains ptr
+uma_memory_pool_handle_t umaPoolByPtr(const void *ptr);
```

Comment:
Yes, I think just returning NULL is fine in that case, I just forgot to write that in the comment. Done.

Reply:
Thinking more about returning NULL... The return type is `uma_memory_pool_handle_t` not `void*`. So that we should define a "null" or "empty" handle.

Reply:
We can do that but we define `uma_memory_pool_handle_t` to be a pointer to structure in the public header anyway so I don't see anything wrong with NULL

Reply:
UR uses plain 'NULL'. Here's an example: https://github.com/oneapi-src/unified-runtime/blob/main/include/ur_api.h#L425

Reply:
Yea, I agree - let's just use NULL. A handle is just a pointer to an opaque type...